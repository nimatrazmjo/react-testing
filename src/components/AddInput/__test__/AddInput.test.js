import { fireEvent, render, screen } from '@testing-library/react';
import AddInput from '../AddInput';
const mockFn = jest.fn();
describe('AddInput', () => {
  it('should render correctly', () => {
    const { container } = render(<AddInput />);
    expect(container).toMatchSnapshot();
  });
  it('Should render the input and button', () => {
    const { getByPlaceholderText, getByText } = render(<AddInput />);
    const input = getByPlaceholderText('Add a new task here...');
    const button = getByText('Add');
    expect(input).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });

  it('SHould be able to type in the input', () => {
    const { getByPlaceholderText } = render(<AddInput />);
    const input = getByPlaceholderText('Add a new task here...');
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue('');
    input.value = 'test';
    expect(input).toHaveValue('test');
  });

  it('Should have empty input when the button is clicked', () => {
    const { getByText, getByPlaceholderText } = render(<AddInput todos={[]} setTodos={mockFn} />);
    const input = getByPlaceholderText(/Add a new task here.../i);
    const button = getByText('Add');
    fireEvent.change(input, { target: { value: 'test' } });
    fireEvent.click(button);
    expect(input.value).toBe('');
  });

});