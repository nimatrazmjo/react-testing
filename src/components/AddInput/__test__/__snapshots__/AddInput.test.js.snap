// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddInput should render correctly 1`] = `
<div>
  <div
    class="input-container"
  >
    <input
      class="input"
      placeholder="Add a new task here..."
      value=""
    />
    <button
      class="add-btn"
    >
      Add
    </button>
  </div>
</div>
`;
